tests/test_views.py

```diff
@@ -632,6 +632,78 @@ def test_detail_view_staff_all_users(self, mock_update):
         assert mock_update.call_count == 0
         assert models.StaleCompletion.objects.filter(resolved=False).count() == 2
 
+    @XBlock.register_temp_plugin(StubCourse, 'course')
+    @XBlock.register_temp_plugin(StubSequential, 'sequential')
+    @XBlock.register_temp_plugin(StubHTML, 'html')
+    def test_detail_view_staff_requested_multiple_users(self):
+        """
+        Test that requesting course completions for a set of users filters out the other enrolled users
+        """
+        version = 1
+        some_user = User.objects.create(username='test_user_2')
+        self.create_enrollment(
+            user=some_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=some_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=3.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
+
+        some_other_user = User.objects.create(username='test_user_3')
+        self.create_enrollment(
+            user=some_other_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=some_other_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=9.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
+
+        yet_another_user = User.objects.create(username='test_user_4')
+        self.create_enrollment(
+            user=yet_another_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=yet_another_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=6.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
```

Comment:
@lgp171188, could you, please, extract this part (with creating the user, creating enrollment and submitting completion) to a helper method (e.g. of `CompletionAPITestMixin`) to simplify the test and to avoid the code repetition?

Reply:
@Agrendalath, I had only added the changes to fix the failing CI checks on Maurier's changes. Will update the PR to address this now.

Reply:
@Agrendalath, I have addressed your review comment in the latest update to this PR. Please check and let me know if the changes are what you had in mind and they are okay.

Reply:
@lgp171188, that looks nice!